azurerm/internal/services/keyvault/tests/key_vault_key_decrypt_data_source_test.go

```diff
@@ -0,0 +1,39 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+)
+
+func TestAccDataSourceAzureRMKeyVaultKeyDecrypt_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_key_vault_key_decrypt", "test")
+	plaintext := "testData"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { acceptance.PreCheck(t) },
+		Providers: acceptance.SupportedProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceKeyVaultKeyDecrypt_basic(data, plaintext),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "plaintext", plaintext),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceKeyVaultKeyDecrypt_basic(data acceptance.TestData, plaintext string) string {
+	t := testAccAzureRMKeyVaultKeyEncrypt_basic(data, plaintext)
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_key_vault_key_decrypt" "test" {
```

Comment:
could we use the encrypt resource and decrypt to check that the workflow works?

Reply:
yes, in this data source, we assign the `cipher_text` of `azurerm_key_vault_key_encrypt` to `payload`, and in the check functions, we checked whether the descrypt value is the inital input plaintext.
In this way, we checked the whole encrypt and decrypt workflow